### PR TITLE
Renamed `DeviceEventFilter` to `DeviceEvents`

### LINKED ISSUE #2591: `DeviceEventFilter` works differently on X11 and Windows
This is likely owed to none of us catching that the initial implementation (#2195) had the documentation be different from the actual behavior of the X11 backend. Specifically, the behavior of `DeviceEventFilter::Always` and `DeviceEventFilter::Never` is flipped WRT the documentation. Due to this, the subsequent Windows implementation (#2409) followed the documentation, thus leading to further inconsistency.

I'm personally of the opinion that the way it works on X11 is the most intuitive, and that the documentation and the Windows backend should be adjusted accordingly. However, I'd like to get the opinion of others before doing anything.

### PR DESCRIPTION
**Pull request by Toniman575**
- [ ] Tested on all platforms changed
- [x] Added an entry to `CHANGELOG.md` if knowledge of this change could be valuable to users
- [x] Updated documentation to reflect any user-facing changes, including notes of platform-specific behavior
- [x] Created or updated an example program if it would help users understand this functionality
- [ ] Updated [feature matrix](https://github.com/rust-windowing/winit/blob/master/FEATURES.md), if new features were added or implemented

Renamed `EventLoopWindowTarget::set_device_event_filter` to `EventLoopWindowTarget::listen_device_events` and `DeviceEventFilter` to `DeviceEvents` and appropriately reversed its variants according to https://github.com/rust-windowing/winit/issues/2591#issuecomment-1353254710.

Could only test it on Linux.

Fixes #2591.

### CODE REVIEW

**Review by madsmtm — APPROVED** [review]
LGTM.

Blocked on https://github.com/rust-windowing/winit/pull/2710

File: src/event_loop.rs
Comment (filnet) [inline src/event_loop.rs:430]: spelling : propagate

File: src/event_loop.rs
Comment (filnet) [inline src/event_loop.rs:432]: spelling : propagate

File: src/platform_impl/linux/mod.rs
Comment (filnet) [inline src/platform_impl/linux/mod.rs:855]: rename `_filter` ?

Comment (filnet) [inline src/platform_impl/linux/mod.rs:855]: should be `set_` something ?

File: src/platform_impl/linux/x11/mod.rs
Comment (filnet) [inline src/platform_impl/linux/x11/mod.rs:544]: to `filter` or not to `filter` ?
seems like names change randomly.

File: src/event_loop.rs
Comment (filnet) [inline src/event_loop.rs:427]: the new name is a bit generic. does not convey what it is or what it is used for.

Comment (daxpedda) [reply src/event_loop.rs:427]: I think @madsmtm original idea was to remove any mention of "filter" to avoid a double negative.
See https://github.com/rust-windowing/winit/issues/2591#issuecomment-1353220718.

Maybe it would be better to follow @ArturKovacs's advice and use `DeviceEventsAllowed` or `DeviceEventsCapture`?

File: src/platform_impl/linux/x11/mod.rs
Comment (daxpedda) [reply src/platform_impl/linux/x11/mod.rs:544]: See https://github.com/rust-windowing/winit/pull/2700#discussion_r1125589786, but the current change should probably remove all mentions of "filter".

Comment (Toniman575) [reply src/platform_impl/linux/x11/mod.rs:544]: Makes sense, I'll change it to `allowed`.

File: src/platform_impl/linux/mod.rs
Comment (Toniman575) [reply src/platform_impl/linux/mod.rs:855]: I don't think in this file that makes sense as it doesn't conflict with other function names.
Happy to change it if a maintainer thinks otherwise.

File: src/event_loop.rs
Comment (Toniman575) [reply src/event_loop.rs:427]: After further review I probably agree with this, would like to wait for a maintainers opinion on this though.

**Comment by Toniman575** [comment]
rebased

**Review by msiglreith — APPROVED** [review]

File: src/event_loop.rs
Comment (kchibisov) [inline src/event_loop.rs:426]:
```suggestion
/// Control when device events are captured.
```

File: CHANGELOG.md
Comment (kchibisov) [inline CHANGELOG.md:12]:
```suggestion
# Unreleased

- **Breaking** Rename `DeviceEventFilter` to `DeviceEvents` reversing the behavior of variants.
- **Breaking** Rename `EventLoopWindowTarget::set_device_event_filter` to `listen_device_events`.
- On X11, fix `EventLoopWindowTarget::listen_device_events` effect being reversed.
```

Comment (kchibisov) [reply CHANGELOG.md:12]: Ensure that there's no empty line with the `**Breaking** below and there's one empty line to section heading.

File: src/event_loop.rs
Comment (kchibisov) [inline src/event_loop.rs:432]: Add `#[default]` for this variant and use `Derive`.

File: src/platform_impl/linux/x11/mod.rs
Comment (kchibisov) [inline src/platform_impl/linux/x11/mod.rs:584]: Could you reverse all of that to `Always` and remove `filter` word all together?

**Review by kchibisov — CHANGES_REQUESTED** [review]

File: src/platform_impl/linux/x11/mod.rs
Comment (Toniman575) [reply src/platform_impl/linux/x11/mod.rs:584]: What do you mean revert all of that to 'Always'?

Comment (kchibisov) [reply src/platform_impl/linux/x11/mod.rs:584]: So we don't have to `!filter_events`? It would be `let device_events = self.device_events.get() == DeviceEvents::Always || (focus && self.device.events.get() == DeviceEvents::WhenFocused)` and then `if device_events`.

File: src/platform_impl/linux/x11/mod.rs
Comment (Toniman575) [reply src/platform_impl/linux/x11/mod.rs]: Yeah my bad, amended it.

File: CHANGELOG.md
Comment (kchibisov) [reply CHANGELOG.md]: The line is too long, I've suggested the splitting in the `suggested change` but it's ignored.

**Pull request merged**
